app/src/main/java/com/duckduckgo/app/browser/httpauth/WebViewHttpAuthStore.kt

```diff
@@ -58,4 +63,8 @@ class RealWebViewHttpAuthStore(
     override fun clearHttpAuthUsernamePassword(webView: WebView) {
         webViewDatabase.clearHttpAuthUsernamePassword()
     }
+
+    override suspend fun cleanHttpAuthDatabase() {
+        databaseCleaner.cleanDatabase()
+    }
```

Comment:
I am struggling with the `databaseCleaner`. It right now is an interface that has only one implementation , the `AuthDatabaseCleaner`. Is this bc before we had also an impl for the app db?

My struggle is that `databaseCleaner.cleanDatabase()` gives me no information about what I am cleaning. So I need to rely on other things like, in this case, its wrapped in a method called `cleanHttpAuthDatabase`. But anyway I feel compelled to go and look how it is injected to see whether we have injected the proper implementation.

I'd suggest to modify the `DatabaseCleaner#cleanDatabase` signature so that it receives the db path. But that relies on the fact that we now only have one db cleaner.
Give that we don't know what will happen with the app db, I would still propose this.


Note: if we agree on this approach, you can forget my comments about passing the db path in the constructor as oppose to passing the `DatabaseLocator`. We'd now pass nothing

Reply:
I ended up passing the db path to the constructor instead of this but no strong opinions.